### ISSUE TITLE
treewide: interpolate src.rev into meta.changelog instead of version 

### ISSUE THREAD

**Issue by lolbinarycat** [NixOS/nixpkgs#299597]
in the process of mass checking nixpkgs metadata url, i see a few very common classes of errors:
1. pointing to `blob/${version}` instead of `blob/v${version}`
2. switching from a stable to unstable versioning scheme, and not updating the changelog url
3. using a more precise versioning scheme than the one used upstream for tags.

all of these would be fixed by interpolating `src.rev` (which will be either a commit or tag) instead of `version` (which often uses a nixpkgs-specific format)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by dotlambda** [comment]
I agree. But this might not work if the changelog refers to GitHub release notes where `refs/tags/v${version}` doesn't work.

**Comment by lolbinarycat** [comment]
hmm, fair point, although i think it should be fine to just not use the fully-qualified ref name in that case? just do `rev = "v${version}"`?

**Comment by lolbinarycat** [comment]
although, release notes won't work for a commit either.  maybe it would be better to just link to `https://github.com/ORG/REPO/releases/` instead of trying to link to a specific release?

although, it will be much less important once nixpkgs metadata is tested automatically.

**Comment by Pandapip1** [comment]
See related discussion at https://github.com/NixOS/nixpkgs/issues/336454
